app/src/main/java/org/eyeseetea/malariacare/fragments/AVFragment.java

```diff
@@ -0,0 +1,158 @@
+    private AVAdapter.ViewType activeViewType = AVAdapter.ViewType.cardview;
+    AVFragment mAVFragment;
+
+    public AVFragment() {
+        mAVFragment = this;
+        this.mMedias = new ArrayList();
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        Log.d(TAG, "onCreate");
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+            Bundle savedInstanceState) {
+        Log.d(TAG, "onCreateView");
+        if (container == null) {
+            return null;
+        }
+        return inflater.inflate(R.layout.av_fragment, container, false);
+    }
+
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        Log.d(TAG, "onActivityCreated");
+        super.onActivityCreated(savedInstanceState);
+
+        initDetailedRecyclerViewList();
+
+        loadMediaListAndAdapter();
+    }
+
+    private void loadMediaListAndAdapter() {
+        IMainExecutor mainExecutor = new UIThreadExecutor();
+        IAsyncExecutor asyncExecutor = new AsyncExecutor();
+
+        GetMediaUseCase getMediaUseCase = new GetMediaUseCase(mainExecutor, asyncExecutor);
+        getMediaUseCase.execute(new GetMediaUseCase.Callback() {
+            @Override
+            public void onSuccess(List<Media> medias) {
+                mMedias=medias;
+                initAdapters();
+            }
+
+            @Override
+            public void onError() {
+                Log.e(TAG, "error getting user program");
+            }
+        });
+    }
+
+    private void initDetailedRecyclerViewList() {
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(mAVFragment.getActivity().getApplicationContext());
+        recyclerViewList = (RecyclerView)  getView().findViewById(R.id.av_recycler);
+        recyclerViewList.setHasFixedSize(true);
+        recyclerViewList.setLayoutManager(linearLayoutManager);
+    }
+
+    /**
+     * Inits adapter.
+     */
+    private void initAdapters() {
+        this.recyclerAdapter = new AVAdapter(this.mMedias, activeViewType, mAVFragment.getActivity().getApplicationContext());
+        recyclerViewList.setAdapter(recyclerAdapter);
+    }
+
+    public void reloadMedia(List<MediaDB> newListMediaDBs) {
+        Log.d(TAG, "reloadMedia (Thread: " + Thread.currentThread().getId() + "): "
+                + newListMediaDBs.size());
+        this.mMedias.clear();
+        this.mMedias.addAll(MediaRepository.fromModel(newListMediaDBs));
+        this.recyclerAdapter.notifyDataSetChanged();
+    }
+
+    public void reloadHeader(Activity activity) {
+
+    }
+
+    @Override
+    public void registerFragmentReceiver() {
+
+    }
+
+    @Override
+    public void unregisterFragmentReceiver() {
+
+    }
+
+    public void reloadData() {
+        loadMediaListAndAdapter();
+    }
+
+    public void toggleLists() {
+        if(activeViewType== AVAdapter.ViewType.cardview){
```

Comment:
functionally the view options are grid view and list view, the enum for view type should has this naming and the code in general, I think detail view and card view is not a correct naming.